### PR TITLE
Update code-gen

### PR DESCRIPTION
**Pull request by saima-s**
## Change Overview

This PR is about updating the code gen to latest.

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test
- [ ] :building_construction: Build

## Issues 

- fixes #issue-number

## Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: pkg/apis/cr/v1alpha1/zz_generated.deepcopy.go [hunk 1]
```diff
@@ -2,7 +2,7 @@
 // +build !ignore_autogenerated
 
 /*
-Copyright 2023 The Kanister Authors.
+Copyright The Kubernetes Authors.
```
Comment (hairyhum) [inline pkg/apis/cr/v1alpha1/zz_generated.deepcopy.go:5]: Is that intentional? Do we want to change the copyright to Kubernetes?

Comment (saima-s) [reply pkg/apis/cr/v1alpha1/zz_generated.deepcopy.go:5]: It came by running `make codegen`. This is by default from package when we re regenerate.

Comment (saima-s) [reply pkg/apis/cr/v1alpha1/zz_generated.deepcopy.go]: Fixed it by adding `boilerplate.go.txt` in repo.

File: build/boilerplate.go.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
-Copyright The Kubernetes Authors.
+Copyright 2023 The Kanister Authors.
```
Comment (ankitjain235) [inline build/boilerplate.go.txt:2]:
```suggestion
Copyright 2024 The Kanister Authors.
```
I suppose we want to use current year?

Comment (saima-s) [reply build/boilerplate.go.txt]: done

**Review by hairyhum — APPROVED** [review]
LGTM

**Review by ankitjain235 — COMMENTED** [review]
LGTM

**Pull request merged**